### PR TITLE
arch: Add max_supported_cores option in smp module

### PR DESCRIPTION
**Pull request by sycamoremoon**
add max_sipported_cores option to allocate __AP_SP size for user to config

### CODE REVIEW

File: src/arch/Mybuild [hunk 1]
```diff
@@ -31,7 +31,9 @@ abstract module syscall { }
 
 @Mandatory
 @DefaultImpl(embox.arch.generic.nousermode)
-abstract module usermode { }
+abstract module usermode {
+	option number max_supported_cores
```
Comment (anton-bondarev) [inline src/arch/Mybuild:35]: Why does it need?

**Comment by sycamoremoon** [comment]
I have corrected the mistake, please review it

**Review by anton-bondarev — APPROVED** [review]
LGTM

**Comment by anton-bondarev** [comment]
@sycamoremoon Thanks
I've aproved the PR.
Please squash the commits during rebasing

**Comment by sycamoremoon** [comment]
HI, I have rebased my commit to upstream/master

**Review by anton-bondarev — APPROVED** [review]
LGTM

**Pull request merged**